packages/autorest.typescript/test/smoke/generated/networkcloud-resource-manager/src/models/index.ts

```diff
@@ -0,0 +1,6251 @@
+  extends coreClient.OperationOptions {}
+
+/** Contains response data for the listNext operation. */
+export type OperationsListNextResponse = OperationListResult;
+
+/** Optional parameters. */
+export interface BareMetalMachinesListBySubscriptionOptionalParams
+  extends coreClient.OperationOptions {}
+
+/** Contains response data for the listBySubscription operation. */
+export type BareMetalMachinesListBySubscriptionResponse = BareMetalMachineList;
+
+/** Optional parameters. */
+export interface BareMetalMachinesListByResourceGroupOptionalParams
+  extends coreClient.OperationOptions {}
+
+/** Contains response data for the listByResourceGroup operation. */
+export type BareMetalMachinesListByResourceGroupResponse = BareMetalMachineList;
+
+/** Optional parameters. */
+export interface BareMetalMachinesGetOptionalParams
+  extends coreClient.OperationOptions {}
+
+/** Contains response data for the get operation. */
+export type BareMetalMachinesGetResponse = BareMetalMachine;
+
+/** Optional parameters. */
+export interface BareMetalMachinesCreateOrUpdateOptionalParams
+  extends coreClient.OperationOptions {
+  /** Delay to wait until next poll, in milliseconds. */
+  updateIntervalInMs?: number;
+  /** A serialized poller which can be used to resume an existing paused Long-Running-Operation. */
+  resumeFrom?: string;
+}
+
+/** Contains response data for the createOrUpdate operation. */
+export type BareMetalMachinesCreateOrUpdateResponse = BareMetalMachine;
+
+/** Optional parameters. */
+export interface BareMetalMachinesDeleteOptionalParams
+  extends coreClient.OperationOptions {
+  /** Delay to wait until next poll, in milliseconds. */
+  updateIntervalInMs?: number;
+  /** A serialized poller which can be used to resume an existing paused Long-Running-Operation. */
+  resumeFrom?: string;
+}
+
+/** Optional parameters. */
+export interface BareMetalMachinesUpdateOptionalParams
+  extends coreClient.OperationOptions {
+  /** The request body. */
+  bareMetalMachineUpdateParameters?: BareMetalMachinePatchParameters;
+  /** Delay to wait until next poll, in milliseconds. */
+  updateIntervalInMs?: number;
+  /** A serialized poller which can be used to resume an existing paused Long-Running-Operation. */
+  resumeFrom?: string;
+}
+
+/** Contains response data for the update operation. */
+export type BareMetalMachinesUpdateResponse = BareMetalMachine;
+
+/** Optional parameters. */
+export interface BareMetalMachinesCordonOptionalParams
+  extends coreClient.OperationOptions {
+  /** The request body. */
+  bareMetalMachineCordonParameters?: BareMetalMachineCordonParameters;
+  /** Delay to wait until next poll, in milliseconds. */
+  updateIntervalInMs?: number;
+  /** A serialized poller which can be used to resume an existing paused Long-Running-Operation. */
+  resumeFrom?: string;
+}
+
+/** Contains response data for the cordon operation. */
+export type BareMetalMachinesCordonResponse = OperationStatusResult;
```

Comment:
With this upgrading we could generate the model in final-state-schema that is `OperationStatusResult`.